client/state/sites/hooks/use-site-global-styles-status.ts

```diff
@@ -1,16 +1,56 @@
 import { useQuery } from '@tanstack/react-query';
 import wpcom from 'calypso/lib/wp';
 import { useSelector } from 'calypso/state';
-import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 import { getSite } from 'calypso/state/sites/selectors';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import type { ExperimentAssignment } from '@automattic/explat-client';
+
+/*
+ * We cannot import `loadExperimentAssignment` directly from 'calypso/lib/explat'
+ * because it runs a side effect that produces an error on SSR contexts.
+ */
+let loadExperimentAssignment = ( experimentName: string ): Promise< ExperimentAssignment > =>
+	new Promise( ( resolve ) =>
+		resolve( { experimentName, variationName: null, retrievedTimestamp: 0, ttl: 0 } )
+	);
+( async () => {
+	if ( typeof window === 'undefined' ) {
+		return;
+	}
+	try {
+		( { loadExperimentAssignment } = await import( 'calypso/lib/explat' ) );
+	} catch ( e ) {}
+} )();
```

Comment:
I find this a bit complicated to follow, what do you think about this approach:

```
let loadExperimentAssignment = ( experimentName: string ): Promise< ExperimentAssignment > =>
	Promise.resolve( {
		experimentName,
		variationName: null,
		retrievedTimestamp: 0,
		ttl: 0,
	} );

if ( typeof window !== 'undefined' ) {
	import( 'calypso/lib/explat' )
		.then( ( module ) => {
			loadExperimentAssignment = module.loadExperimentAssignment;
		} )
		.catch( ( e ) => {
			debug( 'Failed to load the module: ', e );
		} );
}
```

Reply:
Nice! Much simpler now.